packages/react-menu/src/components/MenuDivider/useMenuDivider.ts

```diff
@@ -6,9 +7,12 @@ import type { MenuDividerProps, MenuDividerState } from './MenuDivider.types';
  */
 export const useMenuDivider = (props: MenuDividerProps, ref: React.Ref<HTMLElement>): MenuDividerState => {
   return {
-    ref,
-    role: 'presentation',
-    'aria-hidden': true,
+    root: getNativeElementProps('div', {
+      ref,
+      role: 'presentation',
+      'aria-hidden': true,
+      ...props,
```

Comment:
```suggestion
      role: 'presentation',
      'aria-hidden': true,
      ...props,
      ref,
```

nit: Even if `ref` cannot come from `props`, let's have it last